sw/otbn/crypto/p384_sign.s

```diff
@@ -352,50 +273,61 @@ p384_sign:
 
 /* pointer to k0 (dptr_k0) */
 .globl dptr_k0
+.weak dptr_k0
 dptr_k0:
   .zero 4
 
 /* pointer to k1 (dptr_k1) */
 .globl dptr_k1
+.weak dptr_k1
 dptr_k1:
   .zero 4
 
 /* pointer to msg (dptr_msg) */
 .globl dptr_msg
+.weak dptr_msg
 dptr_msg:
   .zero 4
 
 /* pointer to R (dptr_r) */
 .globl dptr_r
+.weak dptr_r
 dptr_r:
   .zero 4
 
 /* pointer to S (dptr_s) */
 .globl dptr_s
+.weak dptr_s
 dptr_s:
   .zero 4
 
 /* pointer to X (dptr_x) */
 .globl dptr_x
+.weak dptr_x
 dptr_x:
   .zero 4
 
 /* pointer to Y (dptr_y) */
 .globl dptr_y
+.weak dptr_y
 dptr_y:
   .zero 4
 
 /* pointer to d0 (dptr_d0) */
 .globl dptr_d0
+.weak dptr_d0
 dptr_d0:
   .zero 4
 
 /* pointer to d1 (dptr_d1) */
 .globl dptr_d1
+.weak dptr_d1
 dptr_d1:
   .zero 4
 
 /* 704 bytes of scratchpad memory */
 .balign 32
+.globl scratchpad
+.weak scratchpad
```

Comment:
Hmm, is this right? If I'm linking with some other file that has a 200 byte scratchpad (and comes first on the command), I think I'll end up with a 200 byte scratchpad for the `p384_sign` code. Does this symbol really need to be global?

And the same comment for `p384_verify.s`.

Reply:
The problem I encountered was that, when building an OTBN binary including two files with local scratchpads defined in each of them, I ran out of DMEM. My idea then was to make the scratchpad a global symbol to ensure that in the end only a single scratchpad, which is used by all functions, is allocated in DMEM.

I don't know if this is the best solution for it. Maybe there is a better one?

Reply:
Ah, I see. Hmm, I don't think I know anything cleverer. In a previous job, we played around with aliasing regions in the linker script (a sort of manual approach to overlays). But it was really horrible, and hacking on linker scripts is not fun.

Definitely not for this PR, but I wonder whether it would make sense to write this with a "scratchpad.h" that gets included by anything that needs a scratchpad. At least that would mean the "size voting" ends up a bit more explicit (and things would work or break, independent on what else got linked in!)